clang/tools/driver/cc1depscanProtocol.cpp

```diff
@@ -144,21 +144,20 @@ Expected<ScanDaemon> ScanDaemon::launchDaemon(StringRef BasePath,
       BasePathCStr.c_str(),
   };
 
-  const char *ArgsTestMode[] = {
-      Arg0,
-      "-cc1depscand",
-      "-run", // -launch if we want the daemon to fork itself.
-      BasePathCStr.c_str(),
-      "-shutdown",
-  };
-
   static bool LaunchTestDaemon =
       llvm::sys::Process::GetEnv("__CLANG_TEST_CC1DEPSCAND_SHUTDOWN")
           .has_value();
+  static Optional<std::string> LaunchTestArgs =
+      llvm::sys::Process::GetEnv("__CLANG_TEST_CC1DEPSCAND_EXTRA_ARGS");
 
-  ArrayRef<const char *> InitialArgs =
-      LaunchTestDaemon ? makeArrayRef(ArgsTestMode) : makeArrayRef(Args);
+  ArrayRef<const char *> InitialArgs = makeArrayRef(Args);
   SmallVector<const char *> LaunchArgs(InitialArgs.begin(), InitialArgs.end());
+  if (LaunchTestDaemon)
+    LaunchArgs.push_back("-shutdown");
+
+  if (LaunchTestArgs)
+    LaunchArgs.push_back(LaunchTestArgs->data());
```

Comment:
This seems like if you set multiple args with the environment variable, like
```
config.environment['__CLANG_TEST_CC1DEPSCAND_EXTRA_ARGS'] = '-single-command -something-else'
```
it will pass the whole string as one argument and the checks for the arguments will not work, is this correct?

If yes, I'd recommend either to split the args here or switch to checking for a `__CLANG_TEST_CC1DEPSCAND_SINGLE_COMMAND=1` variable and pass the flag when the variable is present.

Reply:
Ah, multiple args was actually my first implementation, then I thought it is probably overkill. I can bring that back (but probably need to have a stringsaver to save split args).

Reply:
I think it should be sufficient to use `StringRef::split()` with the space character, it seems unlikely we'd need something fancier than that (one can just avoid putting unnecessary spaces when setting that variable).